Code/GraphMol/AddHs.cpp

```diff
@@ -435,7 +435,7 @@ void setTerminalAtomCoords(ROMol &mol, unsigned int idx,
               double vol = v1.dotProduct(v2.crossProduct(v3));
               // FIX: this is almost certainly wrong and should use the chiral
               // tag
-              if ((cipCode == "S" && vol < 0) || (cipCode == "R" && vol > 0)) {
+              if ((otherAtom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CCW && vol < 0) || (otherAtom->getChiralTag() == Atom::ChiralType::CHI_TETRAHEDRAL_CW && vol > 0)) {
```

Comment:
The idea here is right, but the implementation doesn't work in the general case.

There is code above (starting at l363) which uses CIP ranks to order the neighbors. The CW/CCW chiral tags, on the other hand, just use the order of the neighbors in the graph.

What I think you can probably do (not tested) is get rid of that `if` and always execute the code from l383-394.